types/await-timeout/index.d.ts

```diff
@@ -7,6 +7,6 @@ declare class Timeout {
     static wrap(promise: Promise<any>, delay: number, error?: string): Promise<any>;
 
     set(delay: number): Promise<undefined>;
-    set(delay: number, message: string): Promise<never>;
+    set(delay: number, message: string | (() => Error) | Error): Promise<never>;
```

Comment:
Thanks, this looks correct!

However, it took me a while to follow the code in the library to be able to see that it's correct. Would it be possible for you to define an intermediate type for `toError` and reference that here? I think that would be helpful since it would make the code here more parallel to that of the library.

If you don't want to, that's okay, just let me know why. The code you have here is probably okay as is too.

Reply:
Thank you for the quick and helpful feedback! I think it's great advice and I will make those adjustments accordingly.